src/snapred/ui/workflow/DiffCalWorkflow.py

```diff
@@ -340,6 +357,54 @@ def _renewFitPeaks(self, peakFunction):
         )
         return self.request(path="calibration/fitpeaks", payload=payload.json())
 
+    @ExceptionToErrLog
+    @Slot(float)
+    def purgeBadPeaks(self, maxChiSq):
+        self._tweakPeakView.disableRecalculateButton()
+        try:
+            # update the max chi sq
+            self.maxChiSq = maxChiSq
+            allPeaks = self.ingredients.groupedPeakLists
+            param_table = mtd[self.fitPeaksDiagnostic].getItem(FitOutputEnum.Parameters.value).toDict()
+            index = param_table["wsindex"]
+            allChi2 = param_table["chi2"]
+            goodPeaks = []
+            for wkspIndex, groupPeaks in enumerate(allPeaks):
+                peaks = groupPeaks.peaks
+                # collect the fit chi-sq parameters for this spectrum, and the fits
+                chi2 = [x2 for i, x2 in zip(index, allChi2) if i == wkspIndex]
+                goodPeaks.append([peak for x2, peak in zip(chi2, peaks) if x2 < maxChiSq])
+            too_fews = [goodPeak for goodPeak in goodPeaks if len(goodPeak) < 2]
+            if too_fews != []:
+                QMessageBox.critical(
+                    self._tweakPeakView,
+                    "Too Few Peaks",
+                    "Purging would result in fewer than the required 2 peaks for calibration.  "
+                    "The current set of peaks will be retained.",
+                    QMessageBox.Ok,
+                )
+            else:
+                for wkspIndex, groupPeaks in enumerate(allPeaks):
+                    groupPeaks.peaks = goodPeaks[wkspIndex]
+                self.peaksWerePurged = True
+
+            # renew the fits to the peaks
+            self._renewFitPeaks(self.peakFunction)
+
+            # update graph with reduced peak list
+            self._tweakPeakView.updateGraphs(
+                self.focusedWorkspace,
+                self.ingredients.groupedPeakLists,
+                self.fitPeaksDiagnostic,
+            )
+
+        except Exception as e:  # noqa BLE001
```

Comment:
could this not be solved by spinning up a worker like https://github.com/neutrons/SNAPRed/blob/523b781248e9dd24164b7c04a6db736cee72cdaf/src/snapred/ui/presenter/WorkflowPresenter.py#L172-L176 ?

Reply:
I can try fixing it in this PR, or I can make a defect to fix it

Reply:
Could I spin a defect to handle fixing both of these?

Reply:
I'd be fine with that, its not pressing